wgpu-hal/src/vulkan/device.rs

```diff
@@ -236,7 +236,7 @@ impl super::DeviceShared {
         &self,
         buffer: &'a super::Buffer,
         ranges: I,
-    ) -> Option<impl 'a + Iterator<Item = vk::MappedMemoryRange>> {
+    ) -> Option<impl 'a + Iterator<Item = vk::MappedMemoryRange<'a>>> {
```

Comment:
> (The compiler suggests the `'_` lifetime but `'a` seemed more appropriate, however, I don't claim to fully understand what's going on here.)

What is going on is that [`vk::MappedMemoryRange` contains a single CPU-side pointer, `p_next`](https://docs.rs/ash/latest/ash/vk/struct.MappedMemoryRange.html), whose lifetime is tracked here.

Since nothing assigns anything borrowed to `p_next`, realistically this should have been `'static` [^1]. However, since it holds a `vk::DeviceMemory` handle in there from `buffer: &'a Buffer` it might make sense to "cheat" and pretend it borrows something from `buffer` :)

[^1]: And `ash` / the Rust compiler allows that, because `::default()` creates a `'static` which will only be specialized when using a builder function that takes a borrow.